libminifi/test/SingleProcessorTestController.h

```diff
@@ -57,6 +58,14 @@ class SingleProcessorTestController : public TestController {
     return trigger();
   }
 
+  auto trigger(std::initializer_list<std::tuple<const std::string_view, std::unordered_map<std::string, std::string>>> flow_files) {
```

Comment:
Use a proper container instead, like `std:array` or `std::vector`.

Use a dedicated type for flow file data representation over a tuple. Could be a dead simple aggregate struct, just give it a name.

Reply:
good idea, I've changed it in https://github.com/apache/nifi-minifi-cpp/pull/1349/commits/fdb1bf2a34423815cfe2cb395e83dcee3bc2f1cd#diff-a019248b6c700d2a0bc69681c6ea011e5afe07861a32a021f992375eeda11847R70

Reply:
If you remove the constructor, then you can use aggregate initialization without having to spell out the type name of the flow file data, and without implicit constructors. The downside is that I think you need to add an empty pair of braces for the attributes every time, but maybe even that is optional if you add a default member initializer.

Reply:
Thanks!

These kinds of issues are only optical, but can't cause real issues. The compiler (or even your editor/IDE) catches them all, early.